### DISCUSSION TITLE
Generation of .eclipseproduct

### DISCUSSION THREAD

**Discussion by ajohnson1** [eclipse-tycho/tycho#733] [General]
The file .eclipseproduct is used in a shared install situation to create a path to the configuration directory under the user's home directory.

>name=Eclipse Platform
>id=org.eclipse.platform
>version=4.23.0

[Multi-user installs](https://help.eclipse.org/latest/topic/org.eclipse.platform.doc.isv/reference/misc/multi_user_installs.html)

This appears to be built by Tycho in eclipse-application, but not by eclipse-repository.
[generateDotEclipseProduct](https://github.com/eclipse/tycho/blob/d13cf32f702e8aa4a88710ffe4eed6984f16c3e0/tycho-packaging-plugin/src/main/java/org/eclipse/tycho/packaging/ProductExportMojo.java#L416)

What is the right way to generate this file?

**Comment by mickaelistria** [comment]
The default [materialize-product](https://www.eclipse.org/tycho/sitedocs/tycho-p2/tycho-p2-director-plugin/materialize-products-mojo.html) goal doesn't seem to provide such option to generate this file. You may want to invoke the Eclipse p2 director with some extra options that allow it to generate that file, or maybe even add this file manually while packaging. Invoking the director with extra options can probably best be achieved best with the [eclipse-run goal](https://www.eclipse.org/tycho/sitedocs/tycho-extras/tycho-eclipserun-plugin/eclipse-run-mojo.html).

**Reply by ajohnson1** [reply]
It looks like the platform just includes the file manually and updates the version number manually for each release.
[features/org.eclipse.platform-feature/rootfiles/.eclipseproduct](https://git.eclipse.org/r/plugins/gitiles/platform/eclipse.platform.releng/+/4dc9387a8d144be3c457a4817e62583f30825271/features/org.eclipse.platform-feature/rootfiles/.eclipseproduct)
